File: Laborator 7/Laborator/2021/MyFakePlayer/app/src/main/java/ro/cunbm/fakeplayer/MainActivity.java
package ro.cunbm.fakeplayer;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity implements
        View.OnClickListener  {

    Button b1,b2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        b1 = (Button) findViewById(R.id.btnStart);
        b1.setOnClickListener(this);

        b2 = (Button) findViewById(R.id.btnStop);
        b2.setOnClickListener(this);

    }

    @Override
    public void onClick(View v) {
        Intent i=new Intent(this, MyPlayerService.class);

        if (v.getId() == R.id.btnStart) {
            i.putExtra(MyPlayerService.EXTRA_PLAYLIST, "main");
            i.putExtra(MyPlayerService.EXTRA_SHUFFLE, true);

            startService(i);
        }
        else {
            stopService(i);
        }
    }
}

File: Laborator 7/Laborator/2024/StopSoundService/app/src/main/java/com/example/stopsoundservice/MainActivity.kt
package com.example.stopsoundservice

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.stopsoundservice.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.btnPlay.setOnClickListener {
            controlService("PLAY")
        }

        binding.btnPause.setOnClickListener {
            controlService("PAUSE")
        }

        binding.btnStop.setOnClickListener {
            controlService("STOP")
        }
    }

    private fun controlService(action: String) {
        val intent = Intent(this, AudioPlayerService::class.java).apply {
            putExtra("ACTION", action)
        }
        startService(intent)
    }

    override fun onDestroy() {
        super.onDestroy()
        // Service-ul rămâne activ, nu îl oprim
    }
}
